balena/models/config.py

```diff
@@ -50,9 +50,11 @@ def get_all(self):
 
         """
 
-        if self._config is None:
+        if not self._config:
             self._config = self.base_request.request(
                 'config', 'GET', endpoint=self.settings.get('api_endpoint'))
```

Comment:
This request will get a 404 from an open source API because there's no `/config` endpoint, and it seems to me you're not handling it.

Reply:
@dfunckt the 404 is handled here https://github.com/balena-io/balena-sdk-python/blob/master/balena/base_request.py#L190-L191.
However, it still breaks since `get_all()` will always throws an exception if open source API is used and users won't be able to fetch the device types. To fix this, I made a small change and `get_all()` will only fetch the configuration not the device types.